tests/components/frontier_silicon/test_config_flow.py

```diff
@@ -415,3 +415,80 @@ async def test_ssdp_nondefault_pin(hass: HomeAssistant) -> None:
 
     assert result["type"] == FlowResultType.ABORT
     assert result["reason"] == "invalid_auth"
+
+
+async def test_reauth_flow(hass: HomeAssistant, config_entry: MockConfigEntry) -> None:
+    """Test reauth flow."""
+    config_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "unique_id": config_entry.unique_id,
+            "entry_id": config_entry.entry_id,
+        },
+        data=config_entry.data,
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "device_config"
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_PIN: "4242"},
+    )
+    assert result2["type"] == FlowResultType.ABORT
+    assert result2["reason"] == "reauth_successful"
+    assert config_entry.data[CONF_PIN] == "4242"
+
+
+@pytest.mark.parametrize(
+    ("friendly_name_error", "result_error"),
+    [
+        (ConnectionError, "cannot_connect"),
+        (InvalidPinException, "invalid_auth"),
+        (ValueError, "unknown"),
+    ],
+)
+async def test_reauth_flow_friendly_name_error(
+    hass: HomeAssistant,
+    friendly_name_error: Exception,
+    result_error: str,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test reauth flow with failures."""
+    config_entry.add_to_hass(hass)
```

Comment:
I think it'd be nice to explicitely check the original value - not necessary, but makes it easier to check the test.
```suggestion
    config_entry.add_to_hass(hass)
    assert config_entry.data[CONF_PIN] == "1234"
```